File: 2256-minimum-average-difference/2256-minimum-average-difference.py
class Solution:
    def minimumAverageDifference(self, nums: List[int]) -> int:
        asum = 0
        for num in nums: asum += num
        curr = 0
        ans = inf
        res = 0
        for i in range(len(nums)):
            curr += nums[i]
            a = (curr//(i+1))
            b = 0
            if i != len(nums)-1:
                b = (asum-curr)//(len(nums)-i-1)
            temp = abs(a - b)
            if temp < ans:
                ans = temp
                res = i
        return res

File: 2256-minimum-average-difference/2256-minimum-average-difference.cpp
class Solution {
public:
    int minimumAverageDifference(vector<int>& nums) {
        long sum = 0;
        for(int i:nums) sum += i;
        long curr = 0;
        long ans = INT_MAX;
        int res = 0;
        for(int i=0; i<nums.size(); i++) {
            curr += nums[i];
            long a = (curr/(i+1));
            long b = 0;
            if(i != nums.size()-1) {
                b = (sum-curr)/(nums.size()-i-1);
            }
            long temp = abs(a - b);
            if(temp < ans) {
                ans = temp;
                res = i;
            }
        }
        return res;
    }
};